src/views/ApartmentCreationView.vue

```diff
@@ -0,0 +1,97 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import { useRouter, useRoute } from 'vue-router';
+import axios from 'axios';
+import ReusableFormComponent from '@/components/ReusableFormComponent.vue';
+
+// Router und Route für Projekt- und Gebäudekontext
+const router = useRouter();
+const route = useRoute();
+const projectId = route.params.projectId as string;
+const parentId = route.query.parentId as string;
+
+// Typ für Felder-Array
+interface Field {
+  name: string;
+  label: string;
+  type: "text" | "textarea" | "checkbox" | "select";
+  options?: any[];
+  required?: boolean;
+  validations?: ((value: any) => string | null)[];
+}
+
+// Form Felder für Apartment Erstellung
+const fields: Field[] = [
+  { name: 'title', type: 'text', label: 'Title', required: true },
+  { name: 'location', type: 'text', label: 'Location', required: true },
+  {
+    name: 'heatingSpace',
+    type: 'text',
+    label: 'Heating Space (m²)',
+    required: true,
+    validations: [(value) => isNaN(Number(value)) ? 'Must be a number' : null],
+  },
+  {
+    name: 'livingSpace',
+    type: 'text',
+    label: 'Living Space (m²)',
+    required: true,
+    validations: [(value) => isNaN(Number(value)) ? 'Must be a number' : null],
+  },
+  { name: 'description', type: 'textarea', label: 'Description', required: false },
+  {
+    name: 'usableSpace',
+    type: 'text',
+    label: 'Usable Space (m²)',
+    required: true,
+    validations: [(value) => isNaN(Number(value)) ? 'Must be a number' : null],
+  },
+];
+
+// Laden-Status für API-Interaktionen
+const isLoading = ref(false);
+
+// Formular-Daten an Backend senden
```

Comment:
For this we use a service. You might want to have a look at the ProjectService and transfer the appartment-functions to a new AppartmentService.ts.